### ISSUE TITLE
Places: Index locations without internet connection

### ISSUE THREAD

**Issue by IceBoosteR** [photoprism/photoprism#1803]
Hey there,
I'm new to Photoprism so I'm not entirely sure if this is a bug or a missing feature, but let me try to explain.

Completely new setup running on docker, totally fresh database.
So I have a bunch of photos in my libary and I was doing an offline scan of my originals (so no internet connection at the time of indexing). After some days, everything was indexed and I was looking into places. Are lot of photos were accuratly placed. No when I look in the map I can drill down into specific places. At some point in time there are like specific places that I want to "visit". Lets say, there are 4 pictures taken at the same place - and I really would like to view them by clicking on the bubbles. However, the only possibility to actually view parts of the pictures is to drill down until I can see every single one. Which is kind of fine, if I would be able to see the picture on full screen. But when I click on a specific picture, the error message "no pictures found" is displayed. And that's kind of why I am opening this ticket to see, if that is a missing feature, or a bug really.

![grafik](https://user-images.githubusercontent.com/37929606/145639123-d1d05b65-c644-4f99-86a3-f544ddba77d0.png)

Also some questions related to this:
a) Would an internet connection have helped on preventing the bug
b) Are the results as accurate as with internet connection

Thanks,
IceBoosteR

**Comment by lastzero** [comment]
Yes, it's possible that locations were not properly indexed without internet connection. We've added a few more validations recently and abort setting a location if an error occurs. They could not just be invalid coordinates but also a failed geodata request. You could have disabled lookups completely though via config option.

**Comment by lastzero** [comment]
Since we sadly have very limited time:

Would you be able to test if it works when the server has a network connection while indexing?

**Comment by lastzero** [comment]
We'll add the bug label when it's a confirmed bug :+1:

**Comment by IceBoosteR** [comment]
Hi,
will run a complete re-index on my library, will take some days to complete because of the size of the data. What I am wondering is, that the location is about to be correct, it's where the pictures has been taken, but I just can't vie them.
Will report back with the result.

**Comment by lastzero** [comment]
You can also try to run the "photoprism places update" command instead of reindexing completely. It could happen because PhotoPrism uses S2 Cell IDs internally and not latitude / longitude directly. Obviously it should still work, but any more detailed analysis takes time.

**Comment by IceBoosteR** [comment]
Ok first I was smart enough to just add new pictures which I have not indexed yet. Indexing with Internet connection did work and now I can view the pictures as it should.

I think I am unable to execute the command you mentioned. Running it docker, connected via bash
`
root@2311e39a2142:/photoprism# photoprism places update
INFO[2021-12-22T20:18:41Z] Help us make a difference and become a sponsor today!
INFO[2021-12-22T20:18:41Z] Visit https://docs.photoprism.app/funding/ to learn more.
WARN[2021-12-22T20:18:41Z] migrate: creating 20211121-094727 failed with UNIQUE constraint failed: migrations.id [980.858µs]
WARN[2021-12-22T20:18:41Z] migrate: creating 20211124-120008 failed with UNIQUE constraint failed: migrations.id [492.245µs]
ERRO[2021-12-22T20:18:43Z] Since running this command puts additional load on our infrastructure, we unfortunately can only offer it to sponsors.
`

**Comment by lastzero** [comment]
Running our latest version? The warnings should be fixed. But yes, this is only available to sponsors just like direct technical support.

**Comment by IceBoosteR** [comment]
`root@2311e39a2142:/photoprism# photoprism --version
PhotoPrism version 211215-93b26f19-Linux-x86_64
`
Thats the version I am running, watchtower did not find any new updates for this container, so I assume it's the latest one

Alright, so I need to reindex the full database unfortunetely then. And is this behaviour to be  taken as a bug? As the coordinates are retrived correctly offline? Its just that the "match" toward opening the correct pisture is missing.
Cheers

**Comment by lastzero** [comment]
Indexing locations without internet connection may be supported when we reach our funding goal. Currently it is not, so it's not a bug technically. That's all I can say or do at 9.52pm while we are out of office for Christmas.

**Comment by IceBoosteR** [comment]
Okay thanks for the info, got you.

(No one pushed you in answering at that time :) )

You can put the ticket in a state how you'd like it for being able to track this feature eventually.

**Comment by lastzero** [comment]
It's not like work is getting less if we wait. Just means doing more work / answering more questions in less time later, which is physically impossible. When possible, we provide quick solutions.
